app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/ExamListAdapter.kt

```diff
@@ -0,0 +1,81 @@
+package de.tum.`in`.tumcampusapp.component.tumui.grades
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.component.other.generic.adapter.SimpleStickyListHeadersAdapter
+import de.tum.`in`.tumcampusapp.component.tumui.grades.model.Exam
+import org.joda.time.format.DateTimeFormat
+
+/**
+ * Custom UI adapter for a list of exams.
+ */
+class ExamListAdapter(context: Context, results: List<Exam>) : SimpleStickyListHeadersAdapter<Exam>(context, results) {
+
+    init {
+        infoList.sort()
+    }
+
+    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
+        val holder: ViewHolder
+        var view: View? = convertView
+
+        if (view == null) {
+            view = mInflater.inflate(R.layout.activity_grades_listview, parent, false)
+            holder = ViewHolder(view!!)
+            view.tag = holder
+        } else {
+            holder = view.tag as ViewHolder
+        }
+
+        val exam = infoList[position]
+        if (exam != null) {
```

Comment:
`exam` should never be `null` here. To avoid warnings, you could use `checkNotNull(infoList[position])`. Then you wouldn’t need `if (exam != null) {...}`.